python/ml4ir/base/features/feature_layer.py

```diff
@@ -87,9 +87,10 @@ def feature_layer_op(inputs):
             feature_tensor = inputs[feature_node_name]
 
             if "fn" in feature_layer_info:
-                feature_tensor = feature_layer_map.get_fn(feature_layer_info["fn"])(
-                    feature_tensor=feature_tensor, feature_info=feature_info, file_io=file_io
-                )
+                feature_fn = feature_layer_map.get_fn(feature_layer_info["fn"])
+                if not feature_fn:
+                    raise RuntimeError("Unsupported feature function: {}".format(feature_layer_info["fn"]))
+                feature_tensor = feature_fn(feature_tensor=feature_tensor, feature_info=feature_info, file_io=file_io)
             elif feature_info["trainable"]:
```

Comment:
why not `else` ?

Reply:
super minor though

Reply:
I disagree with that. 

Here I raise an error in the middle of the function, it will skip the rest of the function.

If I put an else for the next row, it feels like 'either we raise an exception or (else) we assign feature_tensor', when actually the behavior is: raise an exception or continue evaluate the rest of the code (feature_tensor=.. but also everything that would not be in the else statement.

Reply:
My comment was simpler, it does not have anything to do with the exception, the comment was for the outer level of the code. I just suggested to change the `elif` of line 94 to `else` to avoid the pattern if-elif without the else..  I checked again, this was not code you touched though.

Reply:
ha ha, sorry. I did not change it, right.